bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatAccountHandler.java

```diff
@@ -0,0 +1,449 @@
+                .header("X-Zc-Timeout", REQUEST_TIMEOUT).header("X-Zc-Nonce", nonce).header("X-Zc-User-Id", userId)
+                .header("X-Zc-User-Signature", signature).header("X-Zc-Content-Length", "" + reqJson.length());
+
+    }
+
+    private Request addStandardHeadersAndPayload(Request req, AbstractRequest payload)
+            throws UnsupportedEncodingException {
+        requestLogger.listenTo(req);
+
+        return req.header("Connection", "Keep-Alive").header("X-Zc-Major-Domain", "seanywell")
+                .header("X-Zc-Msg-Name", "millService").header("X-Zc-Sub-Domain", "milltype").header("X-Zc-Seq-Id", "1")
+                .header("X-Zc-Version", "1").method(HttpMethod.POST).timeout(5, TimeUnit.SECONDS)
+                .content(new BytesContentProvider((gson.toJson(payload)).getBytes("UTF-8")), CONTENT_TYPE);
+    }
+
+    public void updateRoomTemperature(String roomId, Command command, ModeType mode) {
+        Optional<Home> optionalHome = model.findHomeByRoomId(roomId);
+        Optional<Room> optionalRoom = model.findRoomById(roomId);
+        if (optionalHome.isPresent() && optionalRoom.isPresent()) {
+            SetRoomTempRequest req = new SetRoomTempRequest(optionalHome.get(), optionalRoom.get());
+            if (command instanceof QuantityType<?>) {
+                int newTemp = (int) ((QuantityType<?>) command).longValue();
+                switch (mode) {
+                    case Sleep:
+                        req.sleepTemp = newTemp;
+                        break;
+                    case Away:
+                        req.awayTemp = newTemp;
+                        break;
+                    case Comfort:
+                        req.comfortTemp = newTemp;
+                        break;
+                    default:
+                        logger.info("Cannot set room temp for mode " + mode);
+                }
+                try {
+                    sendLoggedInRequest(req, SetRoomTempResponse.class);
+                } catch (MillheatCommunicationException e) {
+                    logger.info("Error updating temperature for room {}", roomId, e);
+                }
+            } else {
+                logger.info("Error updating temperature for room {}, expected QuantityType but got {}", roomId,
+                        command);
+            }
+        }
+    }
+
+    public void updateIndependentHeaterProperties(@Nullable String macAddress, @Nullable String heaterId,
+            @Nullable Command temperatureCommand, @Nullable Command masterOnOffCommand, @Nullable Command fanCommand) {
+        Optional<Heater> optionalHeater = model.findHeaterByMacOrId(macAddress, heaterId);
+        if (optionalHeater.isPresent()) {
+            Heater heater = optionalHeater.get();
+            int setTemp = heater.getTargetTemp();
+            if (temperatureCommand != null && temperatureCommand instanceof QuantityType<?>) {
```

Comment:
No need to check for null because `instanceof` returns false if `temperatureCommand` is `null`.
```suggestion
            if (temperatureCommand instanceof QuantityType<?>) {
```